invokeai/app/services/invocation_cache/invocation_cache_memory.py

```diff
@@ -93,19 +112,16 @@ def get_status(self) -> InvocationCacheStatus:
             max_size=self._max_cache_size,
         )
 
+    @thread_lock.write
     def _delete_by_match(self, to_match: str) -> None:
         if self._max_cache_size == 0:
             return
-
         keys_to_delete = set()
-        for key, value_tuple in self._cache.items():
-            if to_match in value_tuple[1]:
+        for key, cached_item in self._cache.items():
+            if to_match in cached_item.invocation_output_json:
```

Comment:
@psychedelicious - I am still not convinced I like this logic. If `to_match` for some reason isn't a very unique string, it could cause the wrong nodes to get deleted.

Reply:
The worst case scenario is that the cache is not invalidated when it should be - this results in invocation failures for everything that uses the deleted resource.

I think we need to err on the side of caution here, preferring to clear the cache too much rather than too little.

I couldn't figure out a more general way to identify cache entries from resource identifiers - the outputs are arbitrarily structured objects. Happy for another approach though.